storage/storage_interface.ml

```diff
@@ -995,7 +1005,7 @@ sig
     val attach2 :
       context ->
       dbg: debug_info ->
-      dp: dp -> sr: sr -> vdi: vdi -> read_write: bool -> backend
+      dp: dp -> sr: sr -> vdi: vdi -> vm: vm -> read_write: bool -> backend
```

Comment:
@BenSimsCitrix raised a good point that we cannot change this interface here due to storage migration from older versions. Even having it as optional wouldn't help if the new storage plugin actually requires a correct value here.
So the solution would be to have an `attach3`, `activate3`, `deactivate3`, `detach3` impl.

Reply:
I've added new apis attach3 and activate3, to preserve backwards compatability 
Ive also run the following tests

DiffHostCrossPoolSXM Job ID 2594732 (Old version Lima)
DiffHostCrossPoolSXM Job ID 2594733 (Old version Dundee)